### DISCUSSION TITLE
Unexpected behavior from accumulator op

### DISCUSSION THREAD

**Discussion by positlabs** [cables-gl/cables_docs#252] [! Welcome - readme first!]
I tried out the accumulator op for the first time and noticed that it was doing something strange. The time diff is only being recalculated every time the accumulator is triggered. If the time between triggers is large, then the diff multiplication gives some crazy values. 

I made a modified accumulator to show the behavior that I expect. It recalculates the time diff using requestAnimationFrame. This section of the graph tracks values from key presses and sends it to a slider in the sidebar

![Screen Shot 2022-05-28 at 2 12 19 PM](https://user-images.githubusercontent.com/1918844/170838094-909d3f2d-7fe8-4927-8d60-22fb88fa7e88.png)

https://cables.gl/edit/6kk1nZ

**Comment by steam0r** [comment]
that patch is quite complex, can you provide a simpler example that outlines the problems you have with cables accumulator? using requestanimationframe in your own ops is generally not a good idea and might make other things weird, since cables needs to make sure things run in order. would love to have a look though.

"updating only when triggered" is the right way to do things in cables, most of the time.

**Reply by positlabs** [reply]
Is there some way to get the current delta time? Seems like a pretty common thing to need in patches and shaders, so it would be nice to have exposed as a property in the op, or even CGL. 

I can make a reduced test case for this.

**Reply by positlabs** [reply]
https://cables.gl/p/629634098f32e9e1b9ae1e40?s=02FJHY7kS5THmGZc

**Reply by steam0r** [reply]
so, accumulator adds and multiplies a number and sets the outputvalue _over time_ with the diff being calculated from the last/first time you triggered (so the diff increases even when not triggered). maybe thats where things go wrong?

**Reply by positlabs** [reply]
Yeah exactly. The accumulator is essentially accumulating without executing. I expect the accumulator to only accumulate when I send a trigger to it.
